src/actions/LoginInitActions.tsx

```diff
@@ -173,10 +173,24 @@ const checkSecurityMessages = () => async (
       <SecurityAlertsModal bridge={bridge} messages={relevantMessages} />
     ))
     if (username != null) {
-      dispatch({
-        type: 'NAVIGATE',
-        data: { name: 'passwordLogin', params: { username } }
-      })
+      const info = context.localUsers.find(info => info.username === username)
+      if (info?.pinLoginEnabled) {
```

Comment:
We should check for touch login too, since that's another thing that would be useful on the PIN scene.

```ts
if (info != null && 
  (
    info.pinLoginEnabled ||
    upgradeUser(info, getState().touch).touchLoginEnabled
  )
) {
```